File: backend/src/main/java/fr/pedalons/dto/beta/BetaSignupListResponse.java
package fr.pedalons.dto.beta;

import fr.pedalons.dto.validation.ValidateSchema;
import java.util.List;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

@Schema(description = "Paginated beta sign-up list response")
@ValidateSchema
public record BetaSignupListResponse(
    @Schema(description = "List of sign-ups", required = true) List<BetaSignupDto> signups,
    @Schema(description = "Total number of sign-ups", required = true) long total,
    @Schema(description = "Current page number", required = true) int page,
    @Schema(description = "Page size", required = true) int size) {}

File: frontend/src/api/dto/betaSignupListResponse.ts
import type { BetaSignupDto } from './betaSignupDto.ts'

/**
 * Paginated beta sign-up list response
 */
export interface BetaSignupListResponse {
  /** List of sign-ups */
  signups: BetaSignupDto[]
  /** Total number of sign-ups */
  total: number
  /** Current page number */
  page: number
  /** Page size */
  size: number
}
